Contentful.Core/ContentfulManagementClient.cs

```diff
@@ -426,20 +427,46 @@ public ContentfulManagementClient(HttpClient httpClient, IOptions<ContentfulOpti
         /// Gets all the entries of a space, filtered by an optional querystring. A simpler approach than 
         /// to construct a query manually is to use the <see cref="QueryBuilder{T}"/> class.
         /// </summary>
-        /// <typeparam name="T">The <see cref="IContentfulResource"/> to serialize the response into.</typeparam>
+        /// <typeparam name="T">The type to serialize the response into.</typeparam>
         /// <param name="queryString">The optional querystring to add additional filtering to the query.</param>
         /// <param name="cancellationToken">The optional cancellation token to cancel the operation.</param>
         /// <returns>A <see cref="ContentfulCollection{T}"/> of items.</returns>
         /// <exception cref="ContentfulException">There was an error when communicating with the Contentful API.</exception>
-        public async Task<ContentfulCollection<T>> GetEntriesCollectionAsync<T>(string queryString = null, CancellationToken cancellationToken = default(CancellationToken)) where T : IContentfulResource
+        public async Task<ContentfulCollection<T>> GetEntriesCollectionAsync<T>(string queryString = null, CancellationToken cancellationToken = default(CancellationToken))
         {
             var res = await GetAsync($"{_baseUrl}{_options.SpaceId}/entries{queryString}", cancellationToken).ConfigureAwait(false);
 
             await EnsureSuccessfulResultAsync(res).ConfigureAwait(false);
 
+            var isContentfulResource = typeof(IContentfulResource).GetTypeInfo().IsAssignableFrom(typeof(T).GetTypeInfo());
+
             var jsonObject = JObject.Parse(await res.Content.ReadAsStringAsync().ConfigureAwait(false));
+
             var collection = jsonObject.ToObject<ContentfulCollection<T>>(Serializer);
 
+            if (!isContentfulResource)
```

Comment:
maybe this whole `if` logic could be moved to a `normalizeCollection` function or something similar.